nautobot/ipam/migrations/0031_ipam__prefix__add_parent.py

```diff
@@ -43,6 +46,77 @@ def reverse_it(apps, schema_editor):
         vrf.delete()
 
 
+def migrate_ipaddress_status_to_type(apps, schema_editor):
+    """Migrate IPAddress.status DHCP/SLAAC to types."""
+    Status = apps.get_model("extras", "Status")
+    IPAddress = apps.get_model("ipam", "IPAddress")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    ipaddress_ct = ContentType.objects.get_for_model(IPAddress)
+    statuses = Status.objects.filter(content_types=ipaddress_ct)
+    status_migrated, _ = statuses.get_or_create(
+        name="Migrated",
+        defaults={
+            "color": "ff0000",
+            "description": "DHCP/SLAAC status replaced with `type` of same name by Nautobot 2.0 data migrations.",
+        },
+    )
```

Comment:
```suggestion
    status_migrated, _ = Status.objects.get_or_create(
        name="Migrated",
        defaults={
            "color": "ff0000",
            "description": "DHCP/SLAAC status replaced with `type` of same name by Nautobot 2.0 data migrations.",
        },
    )
    status_migrated.content_types.add(ipaddress_ct)
```

Reply:
Also it might be good to only create this status if it's actually needed (i.e. any DHCP/SLAAC addresses actually exist)?

Reply:
Good idea.